generators/liquibase/generator.ts

```diff
@@ -132,6 +132,13 @@ export default class LiquibaseGenerator extends BaseEntityChangesGenerator {
           prepareFieldForLiquibase(entity, field);
         }
       },
+      validateConsistencyOfField({ entity, field }) {
+        if (field.columnRequired && field.liquibaseDefaultValueAttributeValue) {
+          this.handleCheckFailure(
+            `The field ${field.fieldName} in entity ${entity.name} has both columnRequired and a defaultValue, this can lead to unexpected behaviors.`,
+          );
+        }
```

Comment:
I think the check should be changed to `field.columnRequired && !field.liquibaseDefaultValueAttributeValue`.

Only for new fields probably in
https://github.com/jhipster/generator-jhipster/blob/7695814014621aa693661e89823b5d90ec9d1dc7/generators/liquibase/generator.ts#L191-L193

This can be done later if makes sense.